src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs

```diff
@@ -303,6 +314,36 @@ private void OnIncomingReceiveSucceeded(IncomingReplicationHandler instance)
 
         private void OnSystemDocumentChange(DocumentChangeNotification notification)
         {
+
+            if (notification.Key.Equals(Constants.Replication.DocumentReplicationResolvers,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                DocumentsOperationContext context;
+                using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out context))
+                using (var tx = context.OpenWriteTransaction())
+                {
+                    var scriptConfig = _database.DocumentsStorage.Get(context, Constants.Replication.DocumentReplicationResolvers);
+                    var resolverObj = JsonDeserializationServer.ReplicationManualResolver(scriptConfig.Data);
+
+                    foreach (var kvp in resolverObj.ResolveByCollection)
+                    {
+                        var collection = kvp.Key;
+                        var script = kvp.Value.Script;
+                        if (string.IsNullOrEmpty(script.Trim()))
+                        {
+                            continue;
+                        }
+                        context.DocumentDatabase.DocumentsStorage.AddOrUpdateScript(context, collection, script);
```

Comment:
They _are_ stored in a document, so why also store them again, at non trivial cost, separately?